docs/src/main/asciidoc/kafka.adoc

```diff
@@ -308,6 +308,15 @@ You can build the native executable with:
 ./mvnw package -Pnative
 ----
 
+== Kafka Health Check
+
+If you are using the `quarkus-smallrye-health` extension, `quarkus-kafka` could automatically add a readiness health check
+to validate the connection to the broker. This is disabled by default.
```

Comment:
Why is it disabled by default, wasn't the default to enable it?

Reply:
This is debatable :)

This is a quote from the issue description:
> This is my opinion that as Kafka has built in resiliency for both producer and consumer, the health check should not be enabled by default. Some must have good arguments to enabled it by default and I'm open to discussion

Reply:
I always disable Kafka Health Check from _other frameworks_ that why I propose to disabled it by default in Quarkus ... but I'm OK to enabled it by default if it is not a popular decision :)

Reply:
I always disable Kafka Health Check from other frameworks that why I propose to disabled it by default in Quarkus ... but I'm OK to enabled it by default if it is not a popular decision :)

Reply:
I always disable Kafka Health Check from other frameworks that why I propose to disabled it by default in Quarkus ... but I'm OK to enabled it by default if it is not a popular decision :)

Reply:
I brought it up because it's different than what's suggested for MongoDB, so this caught my attention. I'm not 100% sure either. So what happens if a request comes in which triggers the sending of a Kafka message while the broker isn't accessible. Does this block the incoming request? I.e. does it make sense to have requests routed to that pod or not?

Reply:
It vastely depends of the way you use the `KafkaProducer` see the introduction JavaDoc: https://kafka.apache.org/20/javadoc/org/apache/kafka/clients/producer/KafkaProducer.html

The send method is asynchronous, so if you didn't wait for it inside your HTTP request, it will be proceed.

Depending on the configuration of acknowledge mode, retry and buffer size your message can be batched (so not send immediatly), retried or not. All this can lead to a not direct failure in case of sending a message.

So I would say that in the more common cases, a broker failure will lead to messages being kept inside the buffer of the `KafkaProducer` untill the broker is back and no failures of the incomming HTTP request that uses Kafak, nor any blocking.

As it can varies due to the way the user configure Kafka, it should be the user choice to enable the health check or not.

If the user uses quarkus-smallrye-reactive-messaging-kafka, by default the KafkaSink (https://github.com/smallrye/smallrye-reactive-messaging/blob/master/smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaSink.java) didn't wait for the message to be produces (waitForWriteCompletion is true by default and we didn't override it inside Quarkus) so it should be OK.

Reply:
> didn't wait for the message to be produces (waitForWriteCompletion is true by default and we didn't override it inside Quarkus)

Isn't this self-contradictory?

Reply:
> This is my opinion that as Kafka has built in resiliency for both producer and consumer, the health check should not be enabled by default.

Yes, so based on that I think disabling by default is reasonable.